<?xml version="1.0"?>
<launch>
  <!-- Utility: Move robot in a circle around a table -->
  <!-- Useful for testing pose estimation from different viewing angles -->
  
  <!-- Arguments -->
  <arg name="table_center_x" default="3.0" />
  <arg name="table_center_y" default="4.0" />
  <arg name="circle_radius" default="1.5" />
  <arg name="num_waypoints" default="8" />
  <arg name="num_loops" default="-1" />
  <arg name="wait_at_waypoint" default="5.0" />
  
  <!-- Circle Table Mover Node -->
  <node name="circle_table_mover" 
        pkg="ofpgs_ros" 
        type="circle_table_mover.py" 
        output="screen">
    <param name="table_center_x" value="$(arg table_center_x)" />
    <param name="table_center_y" value="$(arg table_center_y)" />
    <param name="radius" value="$(arg circle_radius)" />
    <param name="num_waypoints" value="$(arg num_waypoints)" />
    <param name="num_loops" value="$(arg num_loops)" />
    <param name="wait_at_waypoint" value="$(arg wait_at_waypoint)" />
  </node>
</launch>
